orchagent/portsorch.cpp

```diff
@@ -245,6 +245,9 @@ PortsOrch::PortsOrch(DBConnector *db, vector<table_name_with_pri_t> &tableNames)
     m_portStatusNotificationConsumer = new swss::NotificationConsumer(notificationsDb, "NOTIFICATIONS");
     auto portStatusNotificatier = new Notifier(m_portStatusNotificationConsumer, this);
     Orch::addExecutor("PORT_STATUS_NOTIFICATIONS", portStatusNotificatier);
+
+    // Try warm start
+    bake();
```

Comment:
Within  PortsOrch::initializePort(Port &p),  for warm start,  oper_status should not be changed.

Reply:
I checked code in void PortsOrch::doPortTask(Consumer &consumer).

During warm start, m_portListLaneMap will have all existing ports from SAI query. So there is no new port created, and no new port initialized.

The function PortsOrch::initializePort(Port &p) will not be called.

---
In reply to: [206311005](https://github.com/Azure/sonic-swss/pull/551#discussion_r206311005) [](ancestors = 206311005)

Reply:
Not sure if there is misunderstanding here,  the warm restart testing I have done so far didn't skip  PortsOrch::initializePort(Port &p).    Even for cold start, m_portListLaneMap will have all the exiting physical ports.

Reply:
I am talking about this code snippet:
```
for (auto it = m_lanesAliasSpeedMap.begin(); it != m_lanesAliasSpeedMap.end();)
{
    bool port_created = false;

    if (m_portListLaneMap.find(it->first) == m_portListLaneMap.end())
    {
```

---
In reply to: [206363163](https://github.com/Azure/sonic-swss/pull/551#discussion_r206363163) [](ancestors = 206363163)

Reply:
This is for port breakout.  PortsOrch::initPort() still will be called.

Reply:
Thanks! Got your idea, and it is really a bug. I am using another PR to fix it. Please help [https://github.com/Azure/sonic-swss/pull/555](https://github.com/Azure/sonic-swss/pull/555)

---
In reply to: [206644401](https://github.com/Azure/sonic-swss/pull/551#discussion_r206644401) [](ancestors = 206644401)